go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -179,7 +161,12 @@ def run_go_get
 
           File.write(tmp_go_file, "package dummypkg\n") unless package
 
-          _, stderr, status = Open3.capture3(ENVIRONMENT, "go get -d")
+          # Use version pinning rather than `latest` just in case
+          # a new version gets released in the middle of our run.
+          version = "v" + dep.version.sub(/^v/i, "")
```

Comment:
Any idea what the `version.sub(/^v/i, "")` is protecting against? Never having written Ruby before, I'm not even quite sure what this is doing... 

I copied it from https://github.com/dependabot/dependabot-core/pull/3590/files#diff-491ccbbf0ae3cfe37dca8fba172104cc5466631d946daf9639deac37b014692eL158 but I noticed it's not used over in https://github.com/dependabot/dependabot-core/blob/7a2085bbee797ba7ffd0c09c83fc4d5481c859c2/go_modules/lib/dependabot/go_modules/update_checker.rb#L74

I'm unclear why the inconsistency between these two. It it a bug or intentional?